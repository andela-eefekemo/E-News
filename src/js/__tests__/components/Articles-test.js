/* global expect jest test */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Article from '../../components/Articles';

test('Article Component', () => {
  const component = mount(<Article />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('Testing Props', () => {
  const wrapper = mount(<Article />);
  expect(wrapper.node.props.title).toEqual('');
  wrapper.setProps({ title: 'the-next-web' });
  expect(wrapper.node.props.title).toEqual('the-next-web');
});

test('Testing props in proper position', () => {
  const wrapper = mount(<Article title="Eguono" description="my name" />);
  const span = wrapper.find('span');
  const p = wrapper.find('p');
  expect(span.text()).toBe('Eguono');
  expect(p.text()).toBe('my name');
});
