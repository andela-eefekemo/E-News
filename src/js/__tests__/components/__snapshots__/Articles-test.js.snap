// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component 1`] = `
<div
  className="col s12 m5"
>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt="Headline"
        src=""
      />
      <span
        className="card-title"
      />
    </div>
    <div
      className="card-content"
    >
      <p />
    </div>
    <div
      className="card-action"
    >
      <a
        href=""
        rel="noopener noreferrer"
        target="_blank"
      >
        View More
      </a>
    </div>
  </div>
</div>
`;
